app/validators/date_validator.rb

```diff
@@ -22,17 +20,13 @@ def date_of_birth_validations(record, attribute, value)
   def date_validations(record, attribute, value)
     return record.errors.add(attribute, :blank_date, article: article(attribute), attribute: humanize(attribute)) if options[:presence] && blank?(value)
 
-    return record.errors.add(attribute, :blank_date_fields, attribute: humanize(attribute), fields: blank_fields(value).to_sentence) if blank_fields(value).any?
+    return record.errors.add(attribute, :blank_date_fields, attribute: humanize(attribute), fields: blank_fields(value).to_sentence) if !blank?(value) && blank_fields(value).any?
 
     return record.errors.add(attribute, invalid_date_locale(options), article: article(attribute), attribute: humanize(attribute)) if is_invalid?(value)
 
     record.errors.add(attribute, :future, article: article(attribute), attribute: humanize(attribute)) if value > Time.zone.today && options[:future]
```

Comment:
By the by, but does this one belong here with the other generic "is this a well-formed date" validations? And should there be a corresponding one to assert that a date should be in the past?

Reply:
It is optional however still part of the main date object validations, so yes I think this is a good place for it.

The validator was created based on existing validations that could be found in the code. As the only validation checks relate to a date having to be in the past were quite specific to the models they weren't a good fit for here. Maybe something to look into a bit more now that the interview wizard is in, as it is the one place that has a past check.